ui/pages/confirmations/hooks/useTypesSignSimulationEnabledInfo.ts

```diff
@@ -0,0 +1,67 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+import { MESSAGE_TYPE } from '../../../../shared/constants/app';
+import { parseTypedDataMessage } from '../../../../shared/modules/transaction.utils';
+import { SignatureRequestType } from '../types/confirm';
+import { isPermitSignatureRequest } from '../utils';
+import { selectUseTransactionSimulations } from '../selectors/preferences';
+import { useConfirmContext } from '../context/confirm';
+
+const NON_PERMIT_SUPPORTED_TYPES_SIGNS = [
+  {
+    domainName: 'Seaport',
+    primaryTypeList: ['BulkOrder'],
+    versionList: ['1.4', '1.5', '1.6'],
+  },
+  {
+    domainName: 'Seaport',
+    primaryTypeList: ['OrderComponents'],
+  },
+];
+
+const isNonPermitSupportedByDecodingAPI = (
+  signatureRequest: SignatureRequestType,
+) => {
+  const data = (signatureRequest as SignatureRequestType).msgParams
+    ?.data as string;
+  if (!data) {
+    return false;
+  }
+  const {
+    domain: { name, version },
+    primaryType,
+  } = parseTypedDataMessage(data);
+  return NON_PERMIT_SUPPORTED_TYPES_SIGNS.some(
+    ({ domainName, primaryTypeList, versionList }) =>
+      name === domainName &&
+      primaryTypeList.includes(primaryType) &&
+      (!versionList || versionList.includes(version)),
+  );
+};
+
+export function useTypesSignSimulationEnabledInfo() {
+  const { currentConfirmation } = useConfirmContext<SignatureRequestType>();
+  const useTransactionSimulations = useSelector(
+    selectUseTransactionSimulations,
```

Comment:
Just confirming we removed the ENV so don't need to check that here?

Reply:
yep